std/range/package.d

```diff
@@ -4625,9 +4625,9 @@ body
 
 /// Ditto
 auto iota(B, E)(B begin, E end)
-if (isFloatingPoint!(CommonType!(B, E)))
+    if (isNumeric!(CommonType!(B, E)))
```

Comment:
I guess you should leave isFloatingPoint. Some lines after there's a version for integer types.